### PR TITLE
refactor(package-manager): abstract the package manager methods

### PR DESCRIPTION
**Pull request by rluvaton**
This abstraction is meant to reduce code duplication and increase code simplicity.
Also, I use `cwd` (_current working directory_) to reduce the redundant `cd <project folder>`.

I'm also thinking of maybe changing from static methods in classes to functions cause all of your classes functions are static which provide no benefit of using class

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ module.exports = {
     '@typescript-eslint/no-non-null-assertion': 0,
     'default-case': 0,
     'import/prefer-default-export': 0,
-    'no-lonely-if': 0
+    'no-lonely-if': 0,
+    'no-shadow': 'off',
+    '@typescript-eslint/no-shadow': ['error'],
```
Comment (rluvaton) [inline .eslintrc.js:33]: Typescript complain about `no-shadow` so I follow [StackOverflow answer](https://stackoverflow.com/a/64745652/5923666) to solve this

**Comment by irevenko** [comment]
😄 First of all thank you for making such a big PR but i don't think thank i'll change the project structure.
🤔 And also Classes here are just for the sake of normalization, nothing more.

**Comment by rluvaton** [comment]
Also, if you want to add support for _pnpm_ you will only need to add implementation and that's it.

**Comment by irevenko** [comment]
I'll work on pnpm and commitlint today!

**Pull request closed without merging**